src/index.js

```diff
@@ -0,0 +1,172 @@
+import UploaderService from './service';
+
+const KILOBYTE = 1000,
+    MEGABYTE = 1024;
+
+class UploadController {
+
+    constructor({url, attachments = {}, maxFiles = 10, headers = {}, optionsObject = {}, updateCb, onLoad, maxFilesText = ''}) {
+        this.attachments = attachments;
+        this.maxFiles = maxFiles;
+        this.url = url;
+        this.optionsObject = optionsObject;
+        this.headers = headers;
+        this.fileCounter = Object.keys(this.attachments).length;
+        this.updateCb = updateCb;
```

Comment:
that's not onUpdate. It is a general callback for updating stuff on the consumer side. Couldn't find a better name for it. I can call it simply 'update' but it doesn't mean much about the fact that it is calling back to the consumer.